hazelcast/src/main/java/com/hazelcast/config/XmlConfigPreProcessor.java

```diff
@@ -97,11 +108,18 @@ private void loadAndReplaceImportElement(Node root, Node node) throws Exception
         root.removeChild(node);
     }
 
+    protected abstract Document parse(InputStream inputStream) throws Exception;
+
+    protected abstract Properties getProperties();
+
+    protected abstract ConfigType getXmlType();
+
```

Comment:
It would be useful to add javadoc for abstract methods. These would help implementor understand the purpose of those methods...